services/retrieve_and_rank/v1.js

```diff
@@ -176,19 +182,232 @@ module.exports = function (RED) {
+            break;
+          case 'delete':
+            if (params.cluster_id && params.config_name) {
+              retrieve_and_rank.deleteConfig(params,function(err,res) {
+                handleWatsonCallback(config.mode,node,msg,err,res);
+              });
+            } else {
+              var message = 'Missing cluster id or config name';
+              node.error(message, msg)
+              return;
+            }
+            break;
+        }
+      });      
+    });
+  }
+
+  function solrCollectionNode(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    this.on('input', function(msg) {
+      setupRankRetrieveNode(msg,config,this,function(retrieve_and_rank) {
+        var params = {
+          cluster_id: config.clusterid,
+          collection_name: config.collectionname
+        }
+        if (!params.cluster_id || !params.collection_name) {
+          var message = 'Missing cluster id or collection name';
+          node.error(message, msg)
+          return;
+        }
+
+        switch (config.mode) {
+          case 'create':
+            if (!config.configname) {
+              var message = 'Missing configuration name';
+              return node.error(message, msg);
+            } else {
+              params.config_name = config.configname;
+              node.status({fill: 'blue', shape: 'ring', text: 'Creating solr collection' });
+              retrieve_and_rank.createCollection(params,function(err,res) {
+                node.status({});
+                handleWatsonCallback(config.mode,node,msg,err,res);
+              });
+            }
+            break;
+          case 'delete':
+            retrieve_and_rank.deleteCollection(params,function(err,res) {
+              handleWatsonCallback(config.mode,node,msg,err,res);
+            });
+            break;
+          case 'index':
+            if (msg.doc && typeof(msg.doc) == 'object') {
+
+              var solrClient = retrieve_and_rank.createSolrClient(params);
+              node.status({fill: 'blue', shape: 'ring', text: 'Indexing document' });
+              solrClient.add(msg.doc,function(err,res) {
+                node.status({});
+                handleWatsonCallback(config.mode,node,msg,err,res,function(){
+                  solrClient.commit(function(err) {
+                    if (err) {
+                      var message = 'Error committing change: ' + err;
+                      return node.error(message, msg);
+                    } else {
+                        msg.payload = 'Document indexed successfully';
+                        node.send(msg);
+                    }
+                  });
+                });
+              });
+            } else {
+              var message = 'Missing or invalid document object';
+              return node.error(message, msg);
+            }
+            break;
+
+          case 'search':
+            if (msg.payload && typeof(msg.payload) == 'string') {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2405715173/issues/source?bid=3225115&fileBranchId=3233651#l393)